### PR TITLE
Add types to auth and auth_blocking.

### PR DESCRIPTION
**Pull request by clokep**
I was poking in this code and got annoyed with not having auto-complete.

I think it is all pretty straightforward, there's a second commit which removes an unused method.

### CODE REVIEW

File: synapse/event_auth.py [hunk 1]
```diff
@@ -688,7 +688,7 @@ def _verify_third_party_invite(event: EventBase, auth_events: StateMap[EventBase
     return False
 
 
-def get_public_keys(invite_event):
+def get_public_keys(invite_event: EventBase) -> List[Dict[str, str]]:
```
Comment (clokep) [inline synapse/event_auth.py:691]: I'm not 100% confident that the values here are strings, but I think they are?

Comment (erikjohnston) [reply synapse/event_auth.py:691]: I think so too, but not 100% sure. I wonder if we want to be saying they are strings if we haven't actually asserted that? The event content is untrusted data after all

Comment (clokep) [reply synapse/event_auth.py:691]: Would `Dict[str, Any]` make more sense?

Comment (erikjohnston) [reply synapse/event_auth.py:691]: Yeah, I'd vote we either assert they're strings or say they're `Any`?

Comment (clokep) [reply synapse/event_auth.py]: Done!

**Review by erikjohnston — APPROVED** [review]

**Pull request merged**